lib/scenic/adapters/postgres/views.rb

```diff
@@ -0,0 +1,56 @@
+module Scenic
+  module Adapters
+    class Postgres
+      # Fetches defined views from the postgres connection.
+      # @api private
+      class Views
+        def initialize(connection)
+          @connection = connection
+        end
+
+        # All of the views that this connection has defined.
+        #
+        # This will include materialized views if those are supported by the
+        # connection.
+        #
+        # @return [Array<Scenic::View>]
+        def all
+          views + materialized_views
+        end
+
+        private
+
+        attr_reader :connection
+
+        def views
+          connection.execute(<<-SQL).map { |result| to_scenic_view(result) }
```

Comment:
I'm in love with this syntax, for whatever I need to do after the heredoc. $0.02.